config/istio/gateway/oauth-envoyfilter.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: mcp-well-known-direct
+  namespace: gateway-system
+spec:
+  workloadSelector:
+    labels:
+      istio: ingressgateway
+  configPatches:
+  - applyTo: HTTP_ROUTE
+    match:
+      context: GATEWAY
+      routeConfiguration:
+        vhost:
+          name: "mcp.127-0-0-1.sslip.io:8080"
+    patch:
+      operation: INSERT_FIRST
+      value:
+        name: mcp-well-known-direct
+        match:
+          path: "/.well-known/oauth-protected-resource/mcp"
+        direct_response:
+          status: 200
+          body:
+            inline_string: >
+              {"resource_name":"MCP Server","resource":"http://mcp.127-0-0-1.sslip.io:8888/mcp","authorization_servers":["http://keycloak.127-0-0-1.sslip.io:8888/realms/mcp"],"bearer_methods_supported": ["header"],"scopes_supported":["basic"]}
```

Comment:
This is pretty ugly. Kinda would prefer it hard coded in the broker than here tbh.  Do we have a follow up issue to capture the required configuration and aggregation of the Oauth values so that it can be properly handled by the broker?  This isn't going to work for kagentai as it is

Reply:
Yes, I agree it's pretty ugly.
At this stage I would prefer it outside the broker than hardcoded in there. 
At least if it's outside (in an EnvoyFilter), it can be tweaked more easily until functionality is added to pass in this config to serve it via the broker.

> Do we have a follow up issue

Once this PR lands, i'll capture what's needed in an issue to bring in this response from broker config.

> This isn't going to work for kagentai as it is

Why is that? I would think it's at least configurable compared to a hardcoded response.
Maybe it doesn't matter if auth is enabled in kagenti only after the follow up?